arbos/block_processor.go

```diff
@@ -191,6 +194,29 @@ func ProduceBlockAdvanced(
 		l1Timestamp:   l1Header.Timestamp,
 	}
 
+	// Espresso-specific validation
+	// TODO test: https://github.com/EspressoSystems/espresso-sequencer/issues/772
+	if chainConfig.Espresso {
```

Comment:
This seems to conflict with what I am currently doing. I'm inclined to think that conducting Espresso validation might not be suitable in this stage. Perhaps we need to discuss this.
https://github.com/EspressoSystems/nitro-espresso-integration/pull/14

Reply:
discussed offline, we will move this logic to ProduceBlock where we have access to all transactions